src/engine/GroupBy.cpp

```diff
@@ -838,30 +838,23 @@ void GroupBy::extractValues(
 
 // _____________________________________________________________________________
 sparqlExpression::VectorWithMemoryLimit<ValueId>
-GroupBy::extractValuesDirectlyFromMap(
-    size_t beginIndex, size_t endIndex,
-    const ad_utility::HashMapWithMemoryLimit<KeyType, ValueType>& map,
-    const std::vector<std::vector<AverageAggregationData>>& aggregationData,
-    IdTable* resultTable, size_t hashMapIndex, size_t outCol) {
-  // Store in a vector such that we can copy it to
-  // `_previousResultsFromSameGroup`.
+GroupBy::getHashMapAggregationResults(
+    IdTable* resultTable, const HashMapAggregationData& aggregationData,
+    size_t dataIndex, size_t beginIndex, size_t endIndex) {
```

Comment:
I think there's some room for optimization, but we can leave that for the next PR that support heterogeneous aggregates.